### PR TITLE
Code conversion experiment 

### PR DESCRIPTION
**Pull request by Raphaeljunior**
I edited the handle function to redirect 10% of the traffic to the node version of error tracker.

### CODE REVIEW

**Comment by Raphaeljunior** [comment]
@jridgewell Please begin the code review. I will edit the URL in once we deploy the node server on Monday.

File: tools/errortracker/errortracker.go [hunk 1]
```diff
@@ -88,6 +87,18 @@ func cloudAuthContext(r *http.Request) (context.Context, error) {
...
+	randomVal := rand.Float64()
+	redirectionRate := 0.1;
+	if randomVal < redirectionRate {
+		client := &http.Client{}
+		req, _ := http.NewRequest("GET", "", nil)
+		req.URL.RawQuery = r.URL.Query().Encode()
+		resp, err := client.Do(req)
+		if err != nil {
+			log.Errorf(c, "Error redirecting to experiment %v", resp)
+		}
```
Comment (cramforce) [inline tools/errortracker/errortracker.go]: How is the URL created?

Comment (Raphaeljunior) [inline tools/errortracker/errortracker.go]: I just edited it in. Its a link to the Node JS project.

File: tools/errortracker/errortracker.go [hunk 2]
```diff
@@ -88,14 +88,25 @@ func cloudAuthContext(r *http.Request) (context.Context, error) {
...
-	logc, err := logging.NewClient(c, appengine.AppID(c), "javascript.errors")
+	randomVal := rand.Float64()
+	redirectionRate := 0.1;
+	if randomVal < redirectionRate {
+		client := &http.Client{}
+		urlString := strings.Replace(r.URL.String(), "amp-error-reporting", "amp-error-reporting-js", 1)
+		req, _ := http.NewRequest("GET", urlString, nil)
+		resp, err := client.Do(req)
+		if err != nil {
+			log.Errorf(c, "Error redirecting to experiment %v", resp)
```
Comment (jridgewell) [inline tools/errortracker/errortracker.go]: Where does this log variable come from?

File: tools/errortracker/errortracker.go [hunk 3]
```diff
@@ -88,14 +88,25 @@ func cloudAuthContext(r *http.Request) (context.Context, error) {
...
+	}
+	logc, err := logging.NewClient(c, ppengine.AppID(c), "javascript.errors")
 	if err != nil {
 		http.Error(w, "Cannot connect to Google Cloud Logging",
 			http.StatusInternalServerError)
 		log.Errorf(c, "Cannot connect to Google Cloud Logging: %v", err)
 		return
 	}
-
+s
```
Comment (jridgewell) [inline tools/errortracker/errortracker.go]: Hm?

File: tools/errortracker/errortracker.go [hunk 4]
```diff
@@ -22,6 +22,7 @@ import (
 	"fmt"
 	"math/rand"
 	"net/http"
+	"net/url"
```
Comment (jridgewell) [inline tools/errortracker/errortracker.go]: Unused?

File: tools/errortracker/errortracker.go (hunk 2 above)
Comment (Raphaeljunior) [inline tools/errortracker/errortracker.go]: Imported at the top "google.golang.org/appengine/log"

File: tools/errortracker/errortracker.go [hunk 5]
```diff
@@ -88,14 +87,24 @@ func cloudAuthContext(r *http.Request) (context.Context, error) {
...
+	if randomVal < redirectionRate {
+		client := &http.Client{}
+		urlString := strings.Replace(r.URL.String(), "amp-error-reporting", "amp-error-reporting-js", 1)
+		req, _ := http.NewRequest("GET", urlString, nil)
+		resp, err := client.Do(req)
+		if err != nil {
+			log.Errorf(c, "Error redirecting to experiment %v", resp)
+		}
+	}
+	logc, err := logging.NewClient(c, ppengine.AppID(c), "javascript.errors")
```
Comment (jridgewell) [inline tools/errortracker/errortracker.go]: Typo here.

**Pull request merged**